python/lsst/rubintv/production/uploaders.py

```diff
@@ -20,37 +20,658 @@
+
+
+def createRemoteS3UploaderForSite():
+    """Create the S3Uploader with the correct config
+       for the site automatically.
+
+    Returns
+    -------
+    uploader : `S3Uploader`
+        The S3Uploader for the site.
+    """
+    site = getSite()
+    match site:
+        case "base":
+            return S3Uploader.from_information(
+                endPoint=EndPoint.USDF,
+                bucket=Bucket.BTS,
+            )
+        case "summit":
+            httpsProxy = 'http://ocio-gpu03.slac.stanford.edu:3128'
+            return S3Uploader.from_information(
+                endPoint=EndPoint.USDF,
+                bucket=Bucket.SUMMIT,
+                httpsProxy=httpsProxy
+            )
+        case "usdf":
+            _LOG.info('No remote uploader is necessary for USDF')
+            return None
+        case "tucson":
+            return S3Uploader.from_information(
+                endPoint=EndPoint.USDF,
+                bucket=Bucket.TTS,
+            )
+        case _:
+            raise ValueError(f"Unknown site: {site}")
+
+
+def checkInstrument(instrument):
+    if instrument not in KNOWN_INSTRUMENTS:
+        raise ValueError(f"Error: {instrument} not in {KNOWN_INSTRUMENTS}")
+
+
+class Bucket(Enum):
+    USDF = 1
+    SUMMIT = 2
+    TTS = 3
+    BTS = 4
+
+
+@dataclass(frozen=True)
+class BucketInformation:
+    profileName: str
+    bucketName: str
+
+
+class EndPoint(Enum):
+    USDF = {
+        "end_point": "https://s3dfrgw.slac.stanford.edu",
+        "buckets_available": {
+            Bucket.SUMMIT: BucketInformation(
+                "rubin-rubintv-data-summit",
+                "rubin-rubintv-data-summit"
+            ),
+            Bucket.USDF: BucketInformation(
+                "rubin-rubintv-data-usdf",
+                "rubin-rubintv-data-usdf"
+            ),
+        },
+    }
+
+    SUMMIT = {
+        "end_point": "https://s3.rubintv.cp.lsst.org",
+        "buckets_available": {
+            Bucket.SUMMIT: BucketInformation("summit-data-summit",
+                                             "rubintv")
+        },
+    }
+
+    BASE = {
+        "end_point": "https://s3.rubintv.ls.lsst.org",
+        "buckets_available": {
+            Bucket.BTS: BucketInformation("base-data-base",
+                                          "rubintv")
+        },
+    }
+
+    TUCSON = {
+        "end_point": "https://s3.rubintv.tu.lsst.org",
+        "buckets_available": {
+            Bucket.TTS: BucketInformation("tucson-data-tucson",
+                                          "rubintv")
+        },
+    }
+
+
+class ConnectionError(Exception):
+    def __init__(self, msg: str):
+        super().__init__(msg)
+
+
+class UploadError(Exception):
+    def __init__(self, msg: str):
+        super().__init__(msg)
+
+
+class IUploader(ABC):
+    """Uploader interface
     """
+
+    def __init__(self):
+        super().__init__()
+
+    @abstractmethod
+    def uploadNightReportData(
```

Comment:
It looks like you ran `black` on an earlier version of the code, but not the latest version. You might want to run it again just so that your function signatures etc. look consistent.